processes_notwindows.go

```diff
@@ -62,28 +63,28 @@ func processesFromProc() ([]ProcStat, error) {
 		pid, err := strconv.Atoi(string(stats[0]))
 		if err != nil {
 			log.Errorf("Failed to convert PID(%s) to int: %s", stats[0], err.Error())
+			continue
 		}
 
-		comm, err := readProcFile(getHostProc() + "/" + string(stats[0]) + "/comm")
+		ppid, err := strconv.Atoi(string(stats[4]))
 		if err != nil {
-			log.Errorf("Failed to read comm(%s): %s", stats[0], err.Error())
+			log.Errorf("Failed to convert PPID(%s) to int: %s", stats[4], err.Error())
 		}
 
-		cmdline, err := readProcFile(getHostProc() + "/" + string(stats[0]) + "/cmdline")
-		if err != nil {
-			log.Errorf("Failed to read cmdline(%s): %s", stats[0], err.Error())
-		}
+		stat := ProcStat{PID: pid, ParentPID: ppid}
 
-		ppid, err := strconv.Atoi(string(stats[4]))
-		if err != nil {
-			log.Errorf("Failed to convert PPID(%s) to int: %s", stats[4], err.Error())
+		comm, err := readProcFile(getHostProc() + "/" + string(stats[0]) + "/comm")
+		if err != nil && err != errorProcessTerminated {
+			log.Errorf("Failed to read comm(%s): %s", stats[0], err.Error())
+		} else if err == nil {
+			stat.Name = string(bytes.TrimRight(comm, "\n"))
 		}
 
-		stat := ProcStat{
-			PID:       pid,
-			ParentPID: ppid,
-			Name:      string(bytes.TrimRight(comm, "\n")),
-			Cmdline:   strings.Replace(string(bytes.TrimRight(cmdline, "\x00")), "\x00", " ", -1),
+		cmdline, err := readProcFile(getHostProc() + "/" + string(stats[0]) + "/cmdline")
```

Comment:
Maybe add a comment to explain this block a little

Reply:
What exactly do you want to explain? 
In prev lines there was 		
`pid, err := strconv.Atoi(string(stats[0]))`
To avoid converting `pid` back to string I used stats[0]

Reply:
ignored because of inactivity